Integrations/QRadar/integration-QRadar.py

```diff
@@ -0,0 +1,971 @@
+        human_readable_trans_assets[human_readable_key] = transform_single_asset_to_hr(populated_asset)
+    # Adding endpoints context items
+    trans_assets['Endpoint'] = endpoint_dict
+    human_readable_trans_assets['Endpoint'] = endpoint_dict
+    return trans_assets, human_readable_trans_assets
+
+
+def transform_single_asset_to_hr(asset):
+    '''
+    Prepares asset for human readable
+    '''
+    hr_asset = []
+    for k, v in asset.iteritems():
+        if isinstance(v, dict):
+            hr_item = v
+            hr_item['Property Name'] = k
+            hr_asset.append(hr_item)
+    return hr_asset
+
+
+def create_single_asset_result_and_enrich_endpoint_dict(asset, endpoint_dict, full_values):
+    asset_dict = {}
+    asset_dict['ID'] = asset['id']
+    for interface in asset['interfaces']:
+        if full_values:
+            endpoint_dict['MACAddress'].append(interface['mac_address'])
+        for ip_address in interface['ip_addresses']:
+            endpoint_dict['IPAddress'].append(ip_address['value'])
+    if full_values:
+        domain_name = get_domain_name(asset['domain_id'])
+        endpoint_dict['Domain'].append(domain_name)
+    # Adding values found in properties of the asset
+    enrich_dict_using_asset_properties(asset, asset_dict, endpoint_dict, full_values)
+    return asset_dict
+
+
+def enrich_dict_using_asset_properties(asset, asset_dict, endpoint_dict, full_values):
+    for prop in asset['properties']:
+        if prop['name'] in ASSET_PROPERTIES_NAMES_MAP:
+            asset_dict[ASSET_PROPERTIES_NAMES_MAP[prop['name']]] = {'Value': prop['value'],
+                                                                    'LastUser': prop['last_reported_by']}
+        elif prop['name'] in ASSET_PROPERTIES_ENDPOINT_NAMES_MAP:
+            endpoint_dict[ASSET_PROPERTIES_ENDPOINT_NAMES_MAP[prop['name']]] = prop['value']
+        elif full_values:
+            if prop['name'] in FULL_ASSET_PROPERTIES_NAMES_MAP:
+                asset_dict[FULL_ASSET_PROPERTIES_NAMES_MAP[prop['name']]] = {'Value': prop['value'],
+                                                                             'LastUser': prop['last_reported_by']}
+    return None
+
+
+# Creates an empty endpoint dictionary (for use in other methods)
+def create_empty_endpoint_dict(full_values):
+    endpoint_dict = {}  # type: dict
+    endpoint_dict['IPAddress'] = []
+    endpoint_dict['OS'] = []
+    if full_values:
+        endpoint_dict['MACAddress'] = []
+        endpoint_dict['Domain'] = []
+    return endpoint_dict
+
+
+# Retrieves domain name using domain id
+def get_domain_name(domain_id):
+    try:
+        query_param = {
+            'query_expression': "SELECT DOMAINNAME({0}) AS 'Domain name' FROM events GROUP BY 'Domain name'".format(
+                domain_id)}
+        search_id = search(query_param)['search_id']
+        return get_search_results(search_id)['events'][0]['Domain name']
+    except Exception:
+        demisto.results({
```

Comment:
please return also the exception str in the warning entry so it will not be hided

Reply:
Fixed